### DISCUSSION TITLE
Import fails because "Dates Conflict"

### DISCUSSION THREAD

**Discussion by GeorgBNM** [plausible/analytics#4878] [Self-Hosted Support] [answered]
Hi Plausible team,

Had to go with an alternative Plausible installation for more than one week for reasons. Now just want to import those numbers into the “normal” Plausible installation (both docker).

The data does not overlap, but Plausible thinks it was so:

“Dates Conflict”
The dates 2024-11-24 to 2024-12-02 overlap with dates we've already imported and cannot be used for new imports.

Any idea, how to combine both data together?

Thanks in advance! 🙏

Best,
Georg

**Comment by ruslandoga** [comment]
👋 @GeorgBNM 

~~You can delete the data for that date range from your "normal" Plausible installation and then it would probably work.~~

**Reply by ruslandoga** [reply]
You can check if there is any data for that date region with queries like these:

```console
$ cd plausible-ce # or wherever you git cloned the repo
$ docker compose exec plausible_events_db clickhouse client -d plausible_events_db
```
```sql
:) select count(*) from events_v2 where timestamp between '2024-11-24' and '2024-12-02';
:) select count(*) from sessions_v2 where start between '2024-11-24' and '2024-12-02';
:) select count(*) from imported_pages where date between '2024-11-24' and '2024-12-02';
-- etc. for other imported_* tables
```

**Reply by ruslandoga** [reply]
> The dates 2024-11-24 to 2024-12-02 overlap with dates we've already imported and cannot be used for new imports.

This warning wouldn't be shown unless there is some imported or "native" data already in the database.

**Reply by GeorgBNM** [reply]
OK @ruslandoga,

I'll try and report here, if it was successful. 🙂

Thanks for the fast response!

**Reply by GeorgBNM** [reply]
Hi @ruslandoga 

Sorry for the very late reply.

Just tried to import after emptying all the tables between the dates.

![image](https://github.com/user-attachments/assets/c6259943-5ae1-408d-886f-c276fd9f5e0b)

Restarting the container does not help at all.

Do you have another solution?

**Reply by ruslandoga** [reply]
Yes :) 

The constraints are actually applied by PostgreSQL, so just clearing ClickHouse tables doesn't help. You would need to update the relevant PostgreSQL table as well. I think it's called `site_imports`

```console
$ cd plausible-ce
$ docker compose exec plausible_db psql -h localhost -U postgres -d plasuible_db
```
```sql
select * from site_imports where site_id = (select id from sites where domain = 'example.com');
-- uptade ...
```

**Reply by GeorgBNM** [reply]
@ruslandoga 🙏

I'll try.

**Reply by GeorgBNM** [reply]
Sadly, this helps not. @ruslandoga 

`plausible_db=# SELECT * FROM site_imports WHERE start_date <= '2024-12-03' AND end_date >= '2024-11-24';
 id | start_date | end_date | source | status | site_id | imported_by_id | inserted_at | updated_at | legacy | label
----+------------+----------+--------+--------+---------+----------------+-------------+------------+--------+-------
(0 rows)`

Same with domains.

Best,
Georg

**Reply by ruslandoga** [reply]
I don't think `start_date <= '2024-12-03' AND end_date >= '2024-11-24'` is needed. It should be easier to list all of the records for the site and just eyeball for the one that includes your date range. And then either update it *carefully* or split it in two to create a "hole" in the imported date range :)

**Reply by GeorgBNM** [reply]
Hi @ruslandoga 

Even with no date / domain and after VACUUM it will not work. ☺️
Never imported any data in Plausible before. That's crazy.

Best,
Georg

**Reply by ruslandoga** [reply]
> Never imported any data in Plausible before. That's crazy.

In https://github.com/plausible/analytics/discussions/4878#discussioncomment-11460148 I mentioned that native (aka not imported) data is also taken into account :)

To speed up the debugging a bit, please share the *exact* results from these queries. And please don't remove anything or add extra date filters or empty any tables ahead of time :)

PostgreSQL:

```console
$ docker compose exec plausible_db psql -h localhost -U postgres -d plausible_db
```

```sql
select id, domain, inserted_at, imported_data, stats_start_date, native_stats_start_at from sites;
select * from site_imports;
```

And ClickHouse:

```console
$ docker compose exec plausible_events_db clickhouse client -d plausible_events_db
```
```sql
set param_site_id=1; -- set it to the id of the relevant site from PostgreSQL sites table
select min(timestamp), max(timestamp) from events_v2 where site_id={site_id:UInt64};
select count(*) from events_v2 where site_id={site_id:UInt64} and timestamp between '2024-11-24' and '2024-12-02';
```

**Reply by GeorgBNM** [reply]
Ok, that changes something! :D

**PostgreSQL**:


```
plausible_db=# select id, domain, inserted_at, imported_data, stats_start_date, native_stats_start_at from sites;
 id |              domain               |     inserted_at     | imported_data | stats_start_date | native_stats_start_at 
----+-----------------------------------+---------------------+---------------+------------------+-----------------------
  2 | bxxxxxxt                          | 2022-07-26 07:40:47 |               | 2022-07-26       | 2022-07-26 07:40:47
  9 | rxxxxxxxxxxxxxxxx.de              | 2024-03-31 13:08:26 |               | 2024-03-31       | 2024-03-31 13:08:27
  8 | sxxxxxxxxxxxxxxx.de               | 2024-03-31 13:01:28 |               | 2024-03-31       | 2024-03-31 13:01:29
  1 | pxxxxxxxxxxxx.de                  | 2022-07-11 07:36:01 |               | 2022-07-11       | 2022-07-11 07:36:01
  7 | bxxxxxxx.de                       | 2022-07-28 17:15:31 |               | 2022-07-31       | 2022-07-28 17:15:31
(5 rows)

 id | start_date | end_date | source | status | site_id | imported_by_id | inserted_at | updated_at | legacy | label
----+------------+----------+--------+--------+---------+----------------+-------------+------------+--------+-------
(0 rows)
```

**Clickhouse**:

```
set param_site_id=2;

SET param_site_id = '2'

Query id: d3814eea-2a54-4b51-bff9-436f0b7335f3

Ok.

0 rows in set. Elapsed: 0.002 sec.

a5cda8e001a2 :) select min(timestamp), max(timestamp) from events_v2 where site_id={site_id:UInt64};

SELECT
    min(timestamp),
    max(timestamp)
FROM events_v2
WHERE site_id = {site_id:UInt64}

Query id: 866df4ee-9a91-4d0c-a5f0-13b7975415a0

   ┌──────min(timestamp)─┬──────max(timestamp)─┐
1. │ 2022-07-26 08:54:06 │ 2025-01-24 12:20:23 │
   └─────────────────────┴─────────────────────┘

1 row in set. Elapsed: 1.318 sec. Processed 173.77 thousand rows, 1.96 MB (131.80 thousand rows/s., 1.49 MB/s.)
Peak memory usage: 131.48 KiB.

select count(*) from events_v2 where site_id={site_id:UInt64} and timestamp between '2024-11-24' and '2024-12-02';

SELECT count(*)
FROM events_v2
WHERE (site_id = {site_id:UInt64}) AND ((timestamp >= '2024-11-24') AND (timestamp <= '2024-12-02'))

Query id: 2e86704e-e5a2-407a-af17-529c272ca97f

   ┌─count()─┐
1. │       0 │
   └─────────┘

1 row in set. Elapsed: 0.058 sec.
```

So it seems, there's still nothing. Right?

Best,
Georg

**Reply by ruslandoga** [reply]
There is `native_stats_start_at=2022-07-26 07:40:47` which prevents any imports for data after 2022-07-26.

Relevant code: https://github.com/plausible/analytics/blob/ab77402f7eced6d31b1d32b8cecf228a3c23d055/lib/plausible/imported.ex#L169-L205

**Reply by GeorgBNM** [reply]
I see.

So, will you solve this “problem” with a future update?

**Reply by ruslandoga** [reply]
I don't know of any plans to change the exports/imports logic right now. Sorry.

**Reply by GeorgBNM** [reply]
Thanks for your effort, @ruslandoga!

But working like this, the import logic seems not feasible, right?

Best,
Georg

**Reply by ruslandoga** [reply, marked as answer]
I agree. But I think the idea was that the "native stats" (i.e. recorded by Plausible) are dominant and the imports are secondary and are only for the data that was recorded before switching to Plausible, like as if someone had been using Google Analytics since 2013 and then switched to Plausible in 2024, they would be able to import 2013 till 2024. The ability to fill in the holes in data that has already been recoded by Plausible (in your case that's all data after 2022-07-26) was probably left out due to possible complications. But that's just my opinion and doesn't represent the position of the core team :)
